src/test/java/com/amazon/ion/impl/EncodingDirectiveCompilationTest.java

```diff
@@ -1589,6 +1590,181 @@ public void taglessExpressionGroup(InputType inputType, StreamType streamType) t
         }
     }
 
+    @ParameterizedTest(name = "{0},{1}")
+    @MethodSource("allCombinations")
+    public void taglessExpressionGroupInDelimitedStruct(InputType inputType, StreamType streamType) throws Exception {
```

Comment:
This test took a fair bit of reading to digest. It looks like it's testing three different scenarios for the same test data, and those with cardinality 4 for InputType/StreamType. 

Not a problem for this PR but we need to find a better way to express these patterns. It may be that JUnit 5 nested tests or maybe something with Kotest would help us reduce the maintenance overhead for tests, because the complexity of grokking these is high.